File: app/src/main/java/se/kicksort/metalarchives/network/Backend.java
package se.kicksort.metalarchives.network;

import android.util.Log;

import java.net.InetAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.net.UnknownHostException;
import java.util.Enumeration;

public class Backend {
    private final static String baseUrl = "https://i79r5qruzb.execute-api.eu-west-1.amazonaws.com/production/";

    public static String getBaseUrl() {
        return baseUrl;
    }

    private static String getLocalIpAddress() throws SocketException, UnknownHostException {
        InetAddress result = null;
        Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
        while (interfaces.hasMoreElements()) {
            Enumeration<InetAddress> addresses = interfaces.nextElement().getInetAddresses();
            while (addresses.hasMoreElements()) {
                InetAddress address = addresses.nextElement();
                if (!address.isLoopbackAddress()) {
                    if (address.isSiteLocalAddress()) {
                        return address.getHostAddress();
                    } else if (result == null) {
                        result = address;
                    }
                }
            }
        }

        return (result != null ? result : InetAddress.getLocalHost()).getHostAddress();
    }
}

File: app/src/main/java/se/kicksort/metalarchives/network/Backend.kt
package se.kicksort.metalarchives.network

import android.util.Log

import java.net.InetAddress
import java.net.NetworkInterface
import java.net.SocketException
import java.net.UnknownHostException
import java.util.Enumeration

object Backend {
    val baseUrl = "http://192.168.1.62:3001/"

    private val localIpAddress: String
        @Throws(SocketException::class, UnknownHostException::class)
        get() {
            var result: InetAddress? = null
            val interfaces = NetworkInterface.getNetworkInterfaces()
            while (interfaces.hasMoreElements()) {
                val addresses = interfaces.nextElement().inetAddresses
                while (addresses.hasMoreElements()) {
                    val address = addresses.nextElement()
                    if (!address.isLoopbackAddress) {
                        if (address.isSiteLocalAddress) {
                            return address.hostAddress
                        } else if (result == null) {
                            result = address
                        }
                    }
                }
            }

            return (if (result != null) result else InetAddress.getLocalHost()).hostAddress
        }
}
